src/components/organisms/WorldEntranceForm/WorldEntranceForm.tsx

```diff
@@ -38,18 +38,23 @@ export interface WorldEntranceFormProps {
   world: WithId<World>;
 }
 
-let codeQuestionsCopy: Question[] | undefined = [];
-let profileQuestionsCopy: Question[] | undefined = [];
-let entranceCopy: EntranceStepConfig[] | undefined = [];
-
 export const WorldEntranceForm: React.FC<WorldEntranceFormProps> = ({
   world,
 }) => {
   const worldId = world.id;
   const { user } = useUser();
 
-  // @debt sync useArray with the form changes or try useFieldArray
   const [isDirtyButtons, setDirtyButtons] = useState(false);
+  const [codeQuestionsList, setCodeQuestionsList] = useState<
```

Comment:
I think "this is where the rabbit lies".

The meaning of "we need to add replace"?

Well if you put `useArray(codeQuestionsList)` instead of ` useArray<Question>(world.questions?.code)`, does't that remove the need of that entire `replace` function thing being part of the hook?